streaming/src/test/scala/org/apache/spark/streaming/StreamingContextSuite.scala

```diff
@@ -297,6 +303,26 @@ class StreamingContextSuite extends SparkFunSuite with BeforeAndAfter with Timeo
     Thread.sleep(100)
   }
 
+  test("registering and de-registering of streamingSource") {
+    val conf = new SparkConf().setMaster(master).setAppName(appName)
+    ssc = new StreamingContext(conf, batchDuration)
+    addInputStream(ssc).register()
+    ssc.start()
+
+    assert(ssc.getState() === StreamingContextState.INITIALIZED)
+    val sources: ArrayBuffer[Source] = StreamingContextSuite.getSources(ssc.env.metricsSystem)
```

Comment:
I don't think it's necessary to explicitly write the type here. It would be simpler not to.